auth/server/src/main/java/com/seagates3/controller/AccountController.java

```diff
@@ -363,10 +363,17 @@ public ServerResponse delete() {
         try {
             root = userDAO.find(account.getName(), "root");
         } catch (DataAccessException e) {
+          LOGGER.error("Failed to find root user for account :" +
+                       account.getName());
             return accountResponseGenerator.internalServerError();
         }
 
-        if (!requestor.getId().equals(root.getId())) {
+        if (root == null || root.getId() == null) {
```

Comment:
is this not caught in line 365, why this is checked again

Reply:
this is basically for handling null id case of root user.
while testing I observed root user might be missing userid .hence I added a extra check just for additional confirmation.